tests/kafkatest/tests/core/replication_test.py

```diff
@@ -122,14 +122,16 @@ def min_cluster_size(self):
     @matrix(failure_mode=["clean_shutdown", "hard_shutdown", "clean_bounce", "hard_bounce"],
             broker_type=["leader"],
             security_protocol=["PLAINTEXT"],
-            enable_idempotence=[True])
+            enable_idempotence=[True],
+            metadata_quorum=quorum.all_non_upgrade)
     @matrix(failure_mode=["clean_shutdown", "hard_shutdown", "clean_bounce", "hard_bounce"],
             broker_type=["leader"],
             security_protocol=["PLAINTEXT", "SASL_SSL"],
             metadata_quorum=quorum.all_non_upgrade)
     @matrix(failure_mode=["clean_shutdown", "hard_shutdown", "clean_bounce", "hard_bounce"],
             broker_type=["controller"],
-            security_protocol=["PLAINTEXT", "SASL_SSL"])
+            security_protocol=["PLAINTEXT", "SASL_SSL"],
+            metadata_quorum=quorum.all_non_upgrade)
```

Comment:
I wonder if this is going to fail.  The below code seems to be incorrect:
```
        if failure_mode == "controller" and metadata_quorum != quorum.zk:
            raise Exception("There is no controller broker when using KRaft metadata quorum")
```

That maybe should be checking for `broker_type` instead of `failure_mode`.

Reply:
Good catch @rondagostino. Looks like this was here as a pre-check to make sure the matrix args were correct. I'll fix this